app/components/Edit/EditSchedule.js

```diff
@@ -16,11 +27,23 @@ function buildSchedule(schedule) {
     Saturday: [{ opens_at: null, closes_at: null, scheduleId }],
     Sunday: [{ opens_at: null, closes_at: null, scheduleId }],
   };
+
   if (schedule) {
     schedule.schedule_days.forEach((curr) => {
-      curr.openChanged=false;
-      curr.closeChanged=false;
-      currSchedule[curr.day] ? currSchedule[curr.day].unshift(curr) : currSchedule[curr.day] = [curr];
+      currDay = curr.day;
+      if (!is24Hours[currDay]) {
+        // Check to see if any of the hour pairs for the day
+        // indicate the resource/service is open 24 hours
+        // if there is a pair only have that in the schedule obj
+        if (curr.opens_at === 0 || curr.closes_at === 2359) {
+          is24Hours[currDay] = true;
+          currSchedule[currDay] = [{ opens_at: 0, closes_at: 2359, id: curr.id }];
```

Comment:
We don't need `scheduleId` here since there is already an `Id` present. We only need `scheduleId` if we are creating a schedule that doesn't already exist. Added a comment to clarify this